# Cluster analysis of two cities, New York and Toronto
### Data Collection
First of all, we will collect postal codes and name of districts(in neighborhood if possible) from wiki as postal code can be utilized as primary key in comparing different dataset from other resources for further process. Using BS4, we will extract key data from wiki webpage directly.

Library Check

In [5]:
import numpy as np
import pandas as pd
import requests
import json
from geopy.geocoders import Nominatim
from pandas.io.json import json_normalize
from matplotlib import pyplot
from sklearn.cluster import KMeans
import folium
from bs4 import BeautifulSoup
print ("Library imported.")

Library imported.


Let's bring json file about NY first.

In [10]:
url="http://cocl.us/new_york_dataset/newyork_data.json"
response = requests.get(url=url).json()

In [11]:
response

{'type': 'FeatureCollection',
 'totalFeatures': 306,
 'features': [{'type': 'Feature',
   'id': 'nyu_2451_34572.1',
   'geometry': {'type': 'Point',
    'coordinates': [-73.84720052054902, 40.89470517661]},
   'geometry_name': 'geom',
   'properties': {'name': 'Wakefield',
    'stacked': 1,
    'annoline1': 'Wakefield',
    'annoline2': None,
    'annoline3': None,
    'annoangle': 0.0,
    'borough': 'Bronx',
    'bbox': [-73.84720052054902,
     40.89470517661,
     -73.84720052054902,
     40.89470517661]}},
  {'type': 'Feature',
   'id': 'nyu_2451_34572.2',
   'geometry': {'type': 'Point',
    'coordinates': [-73.82993910812398, 40.87429419303012]},
   'geometry_name': 'geom',
   'properties': {'name': 'Co-op City',
    'stacked': 2,
    'annoline1': 'Co-op',
    'annoline2': 'City',
    'annoline3': None,
    'annoangle': 0.0,
    'borough': 'Bronx',
    'bbox': [-73.82993910812398,
     40.87429419303012,
     -73.82993910812398,
     40.87429419303012]}},
  {'type': 'Feature',
 

In [12]:
ny_data=response['features']

What we need from this data are names of borough and neighborhood for categorization and geometric coordinates

In [13]:
ny_data[0]

{'type': 'Feature',
 'id': 'nyu_2451_34572.1',
 'geometry': {'type': 'Point',
  'coordinates': [-73.84720052054902, 40.89470517661]},
 'geometry_name': 'geom',
 'properties': {'name': 'Wakefield',
  'stacked': 1,
  'annoline1': 'Wakefield',
  'annoline2': None,
  'annoline3': None,
  'annoangle': 0.0,
  'borough': 'Bronx',
  'bbox': [-73.84720052054902,
   40.89470517661,
   -73.84720052054902,
   40.89470517661]}}

Let's extract 4 columns into new dataframe and rename columns for clarification.

In [14]:
column_names = ['Borough', 'Neighborhood', 'Latitude', 'Longitude']
ny_neighborhoods=pd.DataFrame(columns=column_names)

In [15]:
ny_neighborhoods

Empty DataFrame
Columns: [Borough, Neighborhood, Latitude, Longitude]
Index: []

In [20]:
for data in ny_data:
    borough = data['properties']['borough'] 
    neighborhood_name = data['properties']['name']
        
    neighborhood_latlon = data['geometry']['coordinates']
    neighborhood_lat = neighborhood_latlon[1]
    neighborhood_lon = neighborhood_latlon[0]
    
    ny_neighborhoods = ny_neighborhoods.append({'Borough': borough,
                                          'Neighborhood': neighborhood_name,
                                          'Latitude': neighborhood_lat,
                                          'Longitude': neighborhood_lon}, ignore_index=True)

In [21]:
ny_neighborhoods.shape

(306, 4)

In [22]:
ny_neighborhoods.head(5)

Borough Neighborhood   Latitude  Longitude
0   Bronx    Wakefield  40.894705 -73.847201
1   Bronx   Co-op City  40.874294 -73.829939
2   Bronx  Eastchester  40.887556 -73.827806
3   Bronx    Fieldston  40.895437 -73.905643
4   Bronx    Riverdale  40.890834 -73.912585

We observed that there are 306 neighborhood information in this dataset.

Now, we will information about Toronto from new resources. As 4 columns were selected, we will filter only information we need.

In [23]:
wiki_url="https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M"
raw_data_text=requests.get(wiki_url).text
raw_data=BeautifulSoup(raw_data_text,'lxml')
print("Raw data loaded.")

Raw data loaded.


In [25]:
column_name=['PostalCode','Borough','Neighborhood']
tr_data=pd.DataFrame(columns=column_name)
content=raw_data.find('div',class_='mw-parser-output')
table=content.table.tbody
postalcode=0
borough=0
neighborhood=0

First, we set new dataframe for new data. From data resource, we will find keywords by searching for all pattern keywords in webpage resource.

In [26]:
for tr in table.find_all('tr'):
    count=0
    for td in tr.find_all('td'):
        if count==0:
            postalcode=td.text.strip()
            count=count+1
        elif count==1:
            borough = td.text.strip()
            count=count+1
        elif count==2:
            neighborhood=td.text.strip('\n').replace(']','')
    tr_data=tr_data.append({'PostalCode':postalcode,'Borough':borough,'Neighborhood':neighborhood},ignore_index=True)         

In [29]:
tr_data=tr_data[tr_data.Borough!='Not assigned']
tr_data=tr_data[tr_data.Borough!=0]
tr_data.reset_index(drop=True,inplace=True)

Afer trimming irrelevant data and empty data, we can obtain slimmer dataset without geometric coordinates.

In [30]:
tr_data.head(20)

PostalCode           Borough  \
0         M3A        North York   
1         M4A        North York   
2         M5A  Downtown Toronto   
3         M6A        North York   
4         M7A  Downtown Toronto   
5         M9A         Etobicoke   
6         M1B       Scarborough   
7         M3B        North York   
8         M4B         East York   
9         M5B  Downtown Toronto   
10        M6B        North York   
11        M9B         Etobicoke   
12        M1C       Scarborough   
13        M3C        North York   
14        M4C         East York   
15        M5C  Downtown Toronto   
16        M6C              York   
17        M9C         Etobicoke   
18        M1E       Scarborough   
19        M4E      East Toronto   

                                         Neighborhood  
0                                           Parkwoods  
1                                    Victoria Village  
2                           Regent Park, Harbourfront  
3                    Lawrence Manor, Lawrence Heights  
4         Queen's Park, Ontario Provincial Government  
5             Islington Avenue, Humber Valley Village  
6                                      Malvern, Rouge  
7                                           Don Mills  
8                     Parkview Hill, Woodbine Gardens  
9                            Garden District, Ryerson  
10                                          Glencairn  
11  West Deane Park, Princess Gardens, Martin Grov...  
12             Rouge Hill, Port Union, Highland Creek  
13                                          Don Mills  
14                                   Woodbine Heights  
15                                     St. James Town  
16                                 Humewood-Cedarvale  
17  Eringate, Bloordale Gardens, Old Burnhamthorpe...  
18                  Guildwood, Morningside, West Hill  
19                                        The Beaches

In [31]:
count2=0
for count2 in range(0,tr_data.shape[0]):
    if tr_data.iloc[count2][2]=='Not assigned':
        tr_data.iloc[count2][2]=tr_data.iloc[count2][1]
        count2=+1

In [32]:
df=tr_data.groupby(['PostalCode','Borough'])['Neighborhood'].apply(', '.join).reset_index()

In [33]:
df=df.dropna()
df.describe()

PostalCode     Borough Neighborhood
count         103         103          103
unique        103          10           99
top           M6K  North York    Downsview
freq            1          24            4

In [34]:
empty = 'Not assigned'
df = df[(df.PostalCode != empty ) & (df.Borough != empty) & (df.Neighborhood != empty)]

In [35]:
df.describe()

PostalCode     Borough Neighborhood
count         103         103          103
unique        103          10           99
top           M6K  North York    Downsview
freq            1          24            4

There are 103 neighborhoods in Toronto from this dataset.
Let's attach geometric coordinates on it.

In [41]:
df2=pd.read_csv('Geospatial_Coordinates.csv')
df2.head()

Postal Code   Latitude  Longitude
0         M1B  43.806686 -79.194353
1         M1C  43.784535 -79.160497
2         M1E  43.763573 -79.188711
3         M1G  43.770992 -79.216917
4         M1H  43.773136 -79.239476

In [42]:
df['Latitude']=df2['Latitude']
df['Longitude']=df2['Longitude']
tr_neighborhoods=df

In [43]:
tr_neighborhoods.head(5)

PostalCode      Borough                            Neighborhood   Latitude  \
0        M1B  Scarborough                          Malvern, Rouge  43.806686   
1        M1C  Scarborough  Rouge Hill, Port Union, Highland Creek  43.784535   
2        M1E  Scarborough       Guildwood, Morningside, West Hill  43.763573   
3        M1G  Scarborough                                  Woburn  43.770992   
4        M1H  Scarborough                               Cedarbrae  43.773136   

   Longitude  
0 -79.194353  
1 -79.160497  
2 -79.188711  
3 -79.216917  
4 -79.239476

In [48]:
tr_neighborhoods=tr_neighborhoods.drop('PostalCode',axis=1)
tr_neighborhoods.head(5)

Borough                            Neighborhood   Latitude  Longitude
0  Scarborough                          Malvern, Rouge  43.806686 -79.194353
1  Scarborough  Rouge Hill, Port Union, Highland Creek  43.784535 -79.160497
2  Scarborough       Guildwood, Morningside, West Hill  43.763573 -79.188711
3  Scarborough                                  Woburn  43.770992 -79.216917
4  Scarborough                               Cedarbrae  43.773136 -79.239476

After mearging two dataset and trimming postalcode, we shaped dataset of Toronto in same style as NewYork set.

Foursquare Credentials Import


In [49]:
CLIENT_ID = 'WHV1X2WLZ3U0PXDYKA2EMUWXWLCJFG3GKAYMWDFP4DV0ZGRT' # your Foursquare ID
CLIENT_SECRET = 'OEYXYCRUW0SLABNCORYC5EL4LSDNBRHFKMMIIDARHNBQQ02Z' # your Foursquare Secret
VERSION = '20180605' # Foursquare API version

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: WHV1X2WLZ3U0PXDYKA2EMUWXWLCJFG3GKAYMWDFP4DV0ZGRT
CLIENT_SECRET:OEYXYCRUW0SLABNCORYC5EL4LSDNBRHFKMMIIDARHNBQQ02Z


In [50]:
LIMIT=30
def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

In [51]:
toronto_venues=getNearbyVenues(names=tr_neighborhoods['Neighborhood'],
                                   latitudes=tr_neighborhoods['Latitude'],
                                   longitudes=tr_neighborhoods['Longitude'])

Malvern, Rouge
Rouge Hill, Port Union, Highland Creek
Guildwood, Morningside, West Hill
Woburn
Cedarbrae
Scarborough Village
Kennedy Park, Ionview, East Birchmount Park
Golden Mile, Clairlea, Oakridge
Cliffside, Cliffcrest, Scarborough Village West
Birch Cliff, Cliffside West
Dorset Park, Wexford Heights, Scarborough Town Centre
Wexford, Maryvale
Agincourt
Clarks Corners, Tam O'Shanter, Sullivan
Milliken, Agincourt North, Steeles East, L'Amoreaux East
Steeles West, L'Amoreaux West
Upper Rouge
Hillcrest Village
Fairview, Henry Farm, Oriole
Bayview Village
York Mills, Silver Hills
Willowdale, Newtonbrook
Willowdale, Willowdale East
York Mills West
Willowdale, Willowdale West
Parkwoods
Don Mills
Don Mills
Bathurst Manor, Wilson Heights, Downsview North
Northwood Park, York University
Downsview
Downsview
Downsview
Downsview
Victoria Village
Parkview Hill, Woodbine Gardens
Woodbine Heights
The Beaches
Leaside
Thorncliffe Park
East Toronto, Broadview North (Old East York)
The Danforth West, 

In [52]:
newyork_venues=getNearbyVenues(names=ny_neighborhoods['Neighborhood'],
                                   latitudes=ny_neighborhoods['Latitude'],
                                   longitudes=ny_neighborhoods['Longitude'])

Wakefield
Co-op City
Eastchester
Fieldston
Riverdale
Kingsbridge
Marble Hill
Woodlawn
Norwood
Williamsbridge
Baychester
Pelham Parkway
City Island
Bedford Park
University Heights
Morris Heights
Fordham
East Tremont
West Farms
High  Bridge
Melrose
Mott Haven
Port Morris
Longwood
Hunts Point
Morrisania
Soundview
Clason Point
Throgs Neck
Country Club
Parkchester
Westchester Square
Van Nest
Morris Park
Belmont
Spuyten Duyvil
North Riverdale
Pelham Bay
Schuylerville
Edgewater Park
Castle Hill
Olinville
Pelham Gardens
Concourse
Unionport
Edenwald
Bay Ridge
Bensonhurst
Sunset Park
Greenpoint
Gravesend
Brighton Beach
Sheepshead Bay
Manhattan Terrace
Flatbush
Crown Heights
East Flatbush
Kensington
Windsor Terrace
Prospect Heights
Brownsville
Williamsburg
Bushwick
Bedford Stuyvesant
Brooklyn Heights
Cobble Hill
Carroll Gardens
Red Hook
Gowanus
Fort Greene
Park Slope
Cypress Hills
East New York
Starrett City
Canarsie
Flatlands
Mill Island
Manhattan Beach
Coney Island
Bath Beach
Borough Park
Dyker

In [53]:
toronto_onehot = pd.get_dummies(toronto_venues[['Venue Category']], prefix="", prefix_sep="")

toronto_onehot['Neighborhood'] = toronto_venues['Neighborhood'] 

fixed_columns = [toronto_onehot.columns[-1]] + list(toronto_onehot.columns[:-1])
toronto_onehot = toronto_onehot[fixed_columns]

toronto_onehot.head()

Yoga Studio  Accessories Store  Airport  Airport Food Court  Airport Gate  \
0            0                  0        0                   0             0   
1            0                  0        0                   0             0   
2            0                  0        0                   0             0   
3            0                  0        0                   0             0   
4            0                  0        0                   0             0   

   Airport Lounge  Airport Service  Airport Terminal  American Restaurant  \
0               0                0                 0                    0   
1               0                0                 0                    0   
2               0                0                 0                    0   
3               0                0                 0                    0   
4               0                0                 0                    0   

   Antique Shop  ...  Trail  Train Station  Turkish Restaurant  \
0             0  ...      0              0                   0   
1             0  ...      0              0                   0   
2             0  ...      0              0                   0   
3             0  ...      0              0                   0   
4             0  ...      0              0                   0   

   Vegetarian / Vegan Restaurant  Video Game Store  Vietnamese Restaurant  \
0                              0                 0                      0   
1                              0                 0                      0   
2                              0                 0                      0   
3                              0                 0                      0   
4                              0                 0                      0   

   Warehouse Store  Wine Bar  Wings Joint  Women's Store  
0                0         0            0              0  
1                0         0            0              0  
2                0         0            0              0  
3                0         0            0              0  
4                0         0            0              0  

[5 rows x 234 columns]

In [54]:
newyork_onehot = pd.get_dummies(newyork_venues[['Venue Category']], prefix="", prefix_sep="")

newyork_onehot['Neighborhood'] = newyork_venues['Neighborhood'] 

fixed_columns2 = [newyork_onehot.columns[-1]] + list(newyork_onehot.columns[:-1])
newyork_onehot = newyork_onehot[fixed_columns2]

newyork_onehot.head()

Yoga Studio  Accessories Store  Adult Boutique  Afghan Restaurant  \
0            0                  0               0                  0   
1            0                  0               0                  0   
2            0                  0               0                  0   
3            0                  0               0                  0   
4            0                  0               0                  0   

   African Restaurant  Airport Terminal  American Restaurant  Antique Shop  \
0                   0                 0                    0             0   
1                   0                 0                    0             0   
2                   0                 0                    0             0   
3                   0                 0                    0             0   
4                   0                 0                    0             0   

   Arcade  Arepa Restaurant  ...  Vietnamese Restaurant  Warehouse Store  \
0       0                 0  ...                      0                0   
1       0                 0  ...                      0                0   
2       0                 0  ...                      0                0   
3       0                 0  ...                      0                0   
4       0                 0  ...                      0                0   

   Waste Facility  Waterfront  Weight Loss Center  Whisky Bar  Wine Bar  \
0               0           0                   0           0         0   
1               0           0                   0           0         0   
2               0           0                   0           0         0   
3               0           0                   0           0         0   
4               0           0                   0           0         0   

   Wine Shop  Wings Joint  Women's Store  
0          0            0              0  
1          0            0              0  
2          0            0              0  
3          0            0              0  
4          0            0              0  

[5 rows x 390 columns]

In [55]:
toronto_grouped = toronto_onehot.groupby('Neighborhood').mean().reset_index()
toronto_grouped

Neighborhood  Yoga Studio  \
0                                         Agincourt          0.0   
1                            Alderwood, Long Branch          0.0   
2   Bathurst Manor, Wilson Heights, Downsview North          0.0   
3                                   Bayview Village          0.0   
4                 Bedford Park, Lawrence Manor East          0.0   
..                                              ...          ...   
91                      Willowdale, Willowdale West          0.0   
92                                           Woburn          0.0   
93                                 Woodbine Heights          0.0   
94                                  York Mills West          0.0   
95                         York Mills, Silver Hills          0.0   

    Accessories Store  Airport  Airport Food Court  Airport Gate  \
0                 0.0      0.0                 0.0           0.0   
1                 0.0      0.0                 0.0           0.0   
2                 0.0      0.0                 0.0           0.0   
3                 0.0      0.0                 0.0           0.0   
4                 0.0      0.0                 0.0           0.0   
..                ...      ...                 ...           ...   
91                0.0      0.0                 0.0           0.0   
92                0.0      0.0                 0.0           0.0   
93                0.0      0.0                 0.0           0.0   
94                0.0      0.0                 0.0           0.0   
95                0.0      0.0                 0.0           0.0   

    Airport Lounge  Airport Service  Airport Terminal  American Restaurant  \
0              0.0              0.0               0.0             0.000000   
1              0.0              0.0               0.0             0.000000   
2              0.0              0.0               0.0             0.000000   
3              0.0              0.0               0.0             0.000000   
4              0.0              0.0               0.0             0.045455   
..             ...              ...               ...                  ...   
91             0.0              0.0               0.0             0.000000   
92             0.0              0.0               0.0             0.000000   
93             0.0              0.0               0.0             0.000000   
94             0.0              0.0               0.0             0.000000   
95             0.0              0.0               0.0             0.000000   

    ...  Trail  Train Station  Turkish Restaurant  \
0   ...    0.0            0.0                 0.0   
1   ...    0.0            0.0                 0.0   
2   ...    0.0            0.0                 0.0   
3   ...    0.0            0.0                 0.0   
4   ...    0.0            0.0                 0.0   
..  ...    ...            ...                 ...   
91  ...    0.0            0.0                 0.0   
92  ...    0.0            0.0                 0.0   
93  ...    0.0            0.0                 0.0   
94  ...    0.0            0.0                 0.0   
95  ...    0.0            0.0                 0.0   

    Vegetarian / Vegan Restaurant  Video Game Store  Vietnamese Restaurant  \
0                             0.0               0.0                    0.0   
1                             0.0               0.0                    0.0   
2                             0.0               0.0                    0.0   
3                             0.0               0.0                    0.0   
4                             0.0               0.0                    0.0   
..                            ...               ...                    ...   
91                            0.0               0.0                    0.0   
92                            0.0               0.0                    0.0   
93                            0.0               0.0                    0.0   
94                            0.0               0.0

In [56]:
newyork_grouped = newyork_onehot.groupby('Neighborhood').mean().reset_index()
newyork_grouped

Neighborhood  Yoga Studio  Accessories Store  Adult Boutique  \
0         Allerton          0.0                0.0             0.0   
1         Annadale          0.0                0.0             0.0   
2    Arden Heights          0.0                0.0             0.0   
3        Arlington          0.0                0.0             0.0   
4         Arrochar          0.0                0.0             0.0   
..             ...          ...                ...             ...   
297      Woodhaven          0.0                0.0             0.0   
298       Woodlawn          0.0                0.0             0.0   
299        Woodrow          0.0                0.0             0.0   
300       Woodside          0.0                0.0             0.0   
301      Yorkville          0.0                0.0             0.0   

     Afghan Restaurant  African Restaurant  Airport Terminal  \
0                  0.0                 0.0               0.0   
1                  0.0                 0.0               0.0   
2                  0.0                 0.0               0.0   
3                  0.0                 0.0               0.0   
4                  0.0                 0.0               0.0   
..                 ...                 ...               ...   
297                0.0                 0.0               0.0   
298                0.0                 0.0               0.0   
299                0.0                 0.0               0.0   
300                0.0                 0.0               0.0   
301                0.0                 0.0               0.0   

     American Restaurant  Antique Shop  Arcade  ...  Vietnamese Restaurant  \
0               0.000000           0.0     0.0  ...               0.000000   
1               0.181818           0.0     0.0  ...               0.000000   
2               0.000000           0.0     0.0  ...               0.000000   
3               0.000000           0.0     0.0  ...               0.000000   
4               0.000000           0.0     0.0  ...               0.000000   
..                   ...           ...     ...  ...                    ...   
297             0.000000           0.0     0.0  ...               0.000000   
298             0.000000           0.0     0.0  ...               0.000000   
299             0.000000           0.0     0.0  ...               0.000000   
300             0.100000           0.0     0.0  ...               0.000000   
301             0.000000           0.0     0.0  ...               0.033333   

     Warehouse Store  Waste Facility  Waterfront  Weight Loss Center  \
0                0.0             0.0         0.0                 0.0   
1                0.0             0.0         0.0                 0.0   
2                0.0             0.0         0.0                 0.0   
3                0.0             0.0         0.0                 0.0   
4                0.0             0.0         0.0                 0.0   
..               ...             ...         ...                 ...   
297              0.0             0.0         0.0                 0.0   
298              0.0             0.0         0.0                 0.0   
299              0.0             0.0         0.0                 0.0   
300              0.0             0.0         0.0                 0.0   
301              0.0             0.0         0.0                 0.0   

     Whisky Bar  Wine Bar  Wine Shop  Wings Joint  Women's Store  
0           0.0  0.000000   0.000000          0.0            0.0  
1           0.0  0.000000   0.000000          0.0            0.0  
2           0.0  0.000000   0.000000          0.0            0.0  
3           0.0  0.000000   0.000000          0.0            0.0  
4           0.0  0.000000   0.000000          0.0            0.0  
..          ...       ...        ...          ...            ...  
297         0.0  0.000000   0.000000          0.0            0.0  
298         0.0  0.000000   0.000000          0.0         

In [57]:
toronto_onehot.shape

(1319, 234)

In [58]:
newyork_onehot.shape

(6149, 390)

In [59]:
newyork_grouped.head(10)

Neighborhood  Yoga Studio  Accessories Store  Adult Boutique  \
0         Allerton          0.0                0.0             0.0   
1         Annadale          0.0                0.0             0.0   
2    Arden Heights          0.0                0.0             0.0   
3        Arlington          0.0                0.0             0.0   
4         Arrochar          0.0                0.0             0.0   
5          Arverne          0.0                0.0             0.0   
6          Astoria          0.0                0.0             0.0   
7  Astoria Heights          0.0                0.0             0.0   
8       Auburndale          0.0                0.0             0.0   
9       Bath Beach          0.0                0.0             0.0   

   Afghan Restaurant  African Restaurant  Airport Terminal  \
0                0.0                 0.0               0.0   
1                0.0                 0.0               0.0   
2                0.0                 0.0               0.0   
3                0.0                 0.0               0.0   
4                0.0                 0.0               0.0   
5                0.0                 0.0               0.0   
6                0.0                 0.0               0.0   
7                0.0                 0.0               0.0   
8                0.0                 0.0               0.0   
9                0.0                 0.0               0.0   

   American Restaurant  Antique Shop  Arcade  ...  Vietnamese Restaurant  \
0             0.000000           0.0     0.0  ...                    0.0   
1             0.181818           0.0     0.0  ...                    0.0   
2             0.000000           0.0     0.0  ...                    0.0   
3             0.000000           0.0     0.0  ...                    0.0   
4             0.000000           0.0     0.0  ...                    0.0   
5             0.000000           0.0     0.0  ...                    0.0   
6             0.033333           0.0     0.0  ...                    0.0   
7             0.000000           0.0     0.0  ...                    0.0   
8             0.045455           0.0     0.0  ...                    0.0   
9             0.000000           0.0     0.0  ...                    0.0   

   Warehouse Store  Waste Facility  Waterfront  Weight Loss Center  \
0              0.0             0.0         0.0                 0.0   
1              0.0             0.0         0.0                 0.0   
2              0.0             0.0         0.0                 0.0   
3              0.0             0.0         0.0                 0.0   
4              0.0             0.0         0.0                 0.0   
5              0.0             0.0         0.0                 0.0   
6              0.0             0.0         0.0                 0.0   
7              0.0             0.0         0.0                 0.0   
8              0.0             0.0         0.0                 0.0   
9              0.0             0.0         0.0                 0.0   

   Whisky Bar  Wine Bar  Wine Shop  Wings Joint  Women's Store  
0         0.0       0.0       0.00          0.0            0.0  
1         0.0       0.0       0.00          0.0            0.0  
2         0.0       0.0       0.00          0.0            0.0  
3         0.0       0.0       0.00          0.0            0.0  
4         0.0       0.0       0.00          0.0            0.0  
5         0.0       0.0       0.05          0.0            0.0  
6         0.0       0.0       0.00          0.0            0.0  
7         0.0       0.0       0.00          0.0            0.0  
8         0.0       0.0       0.00          0.0            0.0  
9         0.0       0.0       0.00          0.0            0.0  

[10 rows x 390 columns]

In [60]:
toronto_grouped.head(10)

Neighborhood  Yoga Studio  \
0                                          Agincourt          0.0   
1                             Alderwood, Long Branch          0.0   
2    Bathurst Manor, Wilson Heights, Downsview North          0.0   
3                                    Bayview Village          0.0   
4                  Bedford Park, Lawrence Manor East          0.0   
5                                        Berczy Park          0.0   
6                        Birch Cliff, Cliffside West          0.0   
7       Brockton, Parkdale Village, Exhibition Place          0.0   
8  Business reply mail Processing Centre, South C...          0.0   
9  CN Tower, King and Spadina, Railway Lands, Har...          0.0   

   Accessories Store  Airport  Airport Food Court  Airport Gate  \
0                0.0   0.0000              0.0000        0.0000   
1                0.0   0.0000              0.0000        0.0000   
2                0.0   0.0000              0.0000        0.0000   
3                0.0   0.0000              0.0000        0.0000   
4                0.0   0.0000              0.0000        0.0000   
5                0.0   0.0000              0.0000        0.0000   
6                0.0   0.0000              0.0000        0.0000   
7                0.0   0.0000              0.0000        0.0000   
8                0.0   0.0000              0.0000        0.0000   
9                0.0   0.0625              0.0625        0.0625   

   Airport Lounge  Airport Service  Airport Terminal  American Restaurant  \
0           0.000            0.000            0.0000             0.000000   
1           0.000            0.000            0.0000             0.000000   
2           0.000            0.000            0.0000             0.000000   
3           0.000            0.000            0.0000             0.000000   
4           0.000            0.000            0.0000             0.045455   
5           0.000            0.000            0.0000             0.000000   
6           0.000            0.000            0.0000             0.000000   
7           0.000            0.000            0.0000             0.000000   
8           0.000            0.000            0.0000             0.000000   
9           0.125            0.125            0.0625             0.000000   

   ...  Trail  Train Station  Turkish Restaurant  \
0  ...    0.0            0.0                 0.0   
1  ...    0.0            0.0                 0.0   
2  ...    0.0            0.0                 0.0   
3  ...    0.0            0.0                 0.0   
4  ...    0.0            0.0                 0.0   
5  ...    0.0            0.0                 0.0   
6  ...    0.0            0.0                 0.0   
7  ...    0.0            0.0                 0.0   
8  ...    0.0            0.0                 0.0   
9  ...    0.0            0.0                 0.0   

   Vegetarian / Vegan Restaurant  Video Game Store  Vietnamese Restaurant  \
0                       0.000000               0.0                    0.0   
1                       0.000000               0.0                    0.0   
2                       0.000000               0.0                    0.0   
3                       0.000000               0.0                    0.0   
4                       0.000000               0.0                    0.0   
5                       0.033333               0.0                    0.0   
6                       0.000000               0.0                    0.0   
7                       0.000000               0.0                    0.0   
8                       0.000000               0.0                    0.0   
9                       0.000000               0.0                    0.0   

   Warehouse Store  Wine Bar  Wings Joint  Women's Store  
0              0.0       0.0          0.0            0.0  
1              0.0       0.0          0.0            0.0  
2              0.0       0.0          0.0            0.0  
3              0.0       0.0         

In [61]:
toronto_onehot.head(10)

Yoga Studio  Accessories Store  Airport  Airport Food Court  Airport Gate  \
0            0                  0        0                   0             0   
1            0                  0        0                   0             0   
2            0                  0        0                   0             0   
3            0                  0        0                   0             0   
4            0                  0        0                   0             0   
5            0                  0        0                   0             0   
6            0                  0        0                   0             0   
7            0                  0        0                   0             0   
8            0                  0        0                   0             0   
9            0                  0        0                   0             0   

   Airport Lounge  Airport Service  Airport Terminal  American Restaurant  \
0               0                0                 0                    0   
1               0                0                 0                    0   
2               0                0                 0                    0   
3               0                0                 0                    0   
4               0                0                 0                    0   
5               0                0                 0                    0   
6               0                0                 0                    0   
7               0                0                 0                    0   
8               0                0                 0                    0   
9               0                0                 0                    0   

   Antique Shop  ...  Trail  Train Station  Turkish Restaurant  \
0             0  ...      0              0                   0   
1             0  ...      0              0                   0   
2             0  ...      0              0                   0   
3             0  ...      0              0                   0   
4             0  ...      0              0                   0   
5             0  ...      0              0                   0   
6             0  ...      0              0                   0   
7             0  ...      0              0                   0   
8             0  ...      0              0                   0   
9             0  ...      0              0                   0   

   Vegetarian / Vegan Restaurant  Video Game Store  Vietnamese Restaurant  \
0                              0                 0                      0   
1                              0                 0                      0   
2                              0                 0                      0   
3                              0                 0                      0   
4                              0                 0                      0   
5                              0                 0                      0   
6                              0                 0                      0   
7                              0                 0                      0   
8                              0                 0                      0   
9                              0                 0                      0   

   Warehouse Store  Wine Bar  Wings Joint  Women's Store  
0                0         0            0              0  
1                0         0            0              0  
2                0         0            0              0  
3                0         0            0              0  
4                0         0            0              0  
5                0         0            0              0  
6                0         0            0              0  
7                0         0            0              0  
8                0         0            0              0  
9                0         0            0              0  

[10 rows x 234 columns]

In [62]:
toronto_grouped2 = toronto_onehot.groupby('Neighborhood').sum().reset_index()
toronto_grouped2

Neighborhood  Yoga Studio  \
0                                         Agincourt            0   
1                            Alderwood, Long Branch            0   
2   Bathurst Manor, Wilson Heights, Downsview North            0   
3                                   Bayview Village            0   
4                 Bedford Park, Lawrence Manor East            0   
..                                              ...          ...   
91                      Willowdale, Willowdale West            0   
92                                           Woburn            0   
93                                 Woodbine Heights            0   
94                                  York Mills West            0   
95                         York Mills, Silver Hills            0   

    Accessories Store  Airport  Airport Food Court  Airport Gate  \
0                   0        0                   0             0   
1                   0        0                   0             0   
2                   0        0                   0             0   
3                   0        0                   0             0   
4                   0        0                   0             0   
..                ...      ...                 ...           ...   
91                  0        0                   0             0   
92                  0        0                   0             0   
93                  0        0                   0             0   
94                  0        0                   0             0   
95                  0        0                   0             0   

    Airport Lounge  Airport Service  Airport Terminal  American Restaurant  \
0                0                0                 0                    0   
1                0                0                 0                    0   
2                0                0                 0                    0   
3                0                0                 0                    0   
4                0                0                 0                    1   
..             ...              ...               ...                  ...   
91               0                0                 0                    0   
92               0                0                 0                    0   
93               0                0                 0                    0   
94               0                0                 0                    0   
95               0                0                 0                    0   

    ...  Trail  Train Station  Turkish Restaurant  \
0   ...      0              0                   0   
1   ...      0              0                   0   
2   ...      0              0                   0   
3   ...      0              0                   0   
4   ...      0              0                   0   
..  ...    ...            ...                 ...   
91  ...      0              0                   0   
92  ...      0              0                   0   
93  ...      0              0                   0   
94  ...      0              0                   0   
95  ...      0              0                   0   

    Vegetarian / Vegan Restaurant  Video Game Store  Vietnamese Restaurant  \
0                               0                 0                      0   
1                               0                 0                      0   
2                               0                 0                      0   
3                               0                 0                      0   
4                               0                 0                      0   
..                            ...               ...                    ...   
91                              0                 0                      0   
92                              0                 0                      0   
93                              0                 0                      0   
94                              0                 0

In [63]:
newyork_grouped2 = newyork_onehot.groupby('Neighborhood').sum().reset_index()
newyork_grouped2

Neighborhood  Yoga Studio  Accessories Store  Adult Boutique  \
0         Allerton            0                  0               0   
1         Annadale            0                  0               0   
2    Arden Heights            0                  0               0   
3        Arlington            0                  0               0   
4         Arrochar            0                  0               0   
..             ...          ...                ...             ...   
297      Woodhaven            0                  0               0   
298       Woodlawn            0                  0               0   
299        Woodrow            0                  0               0   
300       Woodside            0                  0               0   
301      Yorkville            0                  0               0   

     Afghan Restaurant  African Restaurant  Airport Terminal  \
0                    0                   0                 0   
1                    0                   0                 0   
2                    0                   0                 0   
3                    0                   0                 0   
4                    0                   0                 0   
..                 ...                 ...               ...   
297                  0                   0                 0   
298                  0                   0                 0   
299                  0                   0                 0   
300                  0                   0                 0   
301                  0                   0                 0   

     American Restaurant  Antique Shop  Arcade  ...  Vietnamese Restaurant  \
0                      0             0       0  ...                      0   
1                      2             0       0  ...                      0   
2                      0             0       0  ...                      0   
3                      0             0       0  ...                      0   
4                      0             0       0  ...                      0   
..                   ...           ...     ...  ...                    ...   
297                    0             0       0  ...                      0   
298                    0             0       0  ...                      0   
299                    0             0       0  ...                      0   
300                    3             0       0  ...                      0   
301                    0             0       0  ...                      1   

     Warehouse Store  Waste Facility  Waterfront  Weight Loss Center  \
0                  0               0           0                   0   
1                  0               0           0                   0   
2                  0               0           0                   0   
3                  0               0           0                   0   
4                  0               0           0                   0   
..               ...             ...         ...                 ...   
297                0               0           0                   0   
298                0               0           0                   0   
299                0               0           0                   0   
300                0               0           0                   0   
301                0               0           0                   0   

     Whisky Bar  Wine Bar  Wine Shop  Wings Joint  Women's Store  
0             0         0          0            0              0  
1             0         0          0            0              0  
2             0         0          0            0              0  
3             0         0          0            0              0  
4             0         0          0            0              0  
..          ...       ...        ...          ...            ...  
297           0         0          0            0              0  
298           0         0          0            0         